src/uu/top/src/top.rs

```diff
@@ -0,0 +1,272 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use std::{thread::sleep, time::Duration};
+
+use clap::{arg, crate_version, value_parser, ArgAction, ArgGroup, ArgMatches, Command};
+use picker::sysinfo;
+use prettytable::{format::consts::FORMAT_CLEAN, Row, Table};
+use sysinfo::{Pid, Users};
+use uucore::{
+    error::{UResult, USimpleError},
+    format_usage, help_about, help_usage,
+};
+
+const ABOUT: &str = help_about!("top.md");
+const USAGE: &str = help_usage!("top.md");
+
+mod field;
+mod picker;
+
+#[allow(unused)]
+#[derive(Debug)]
+enum Filter {
+    Pid(Vec<u32>),
+    User(String),
+    EUser(String),
+}
+
+#[derive(Debug)]
+struct Settings {
+    // batch:bool
+    filter: Option<Filter>,
+    width: Option<usize>,
+}
+
+impl Settings {
+    fn new(matches: &ArgMatches) -> UResult<Self> {
+        let width = matches.get_one::<usize>("width").cloned();
+
+        Ok(Self {
+            width,
+            filter: None,
+        })
+    }
+}
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    // Must refresh twice.
+    // https://docs.rs/sysinfo/0.31.2/sysinfo/struct.System.html#method.refresh_cpu_usage
+    picker::sysinfo().write().unwrap().refresh_all();
+    sleep(Duration::from_millis(200));
+    picker::sysinfo().write().unwrap().refresh_all();
+
+    let settings = Settings::new(&matches)?;
+
+    let settings = {
+        let filter = matches
+            .get_many::<u32>("pid")
+            .map(|pidlist| Filter::Pid(pidlist.cloned().collect()))
+            .or_else(|| {
+                matches
+                    .get_one::<String>("filter-any-user")
+                    .map(|user| Filter::User(user.clone()))
+            })
+            .or_else(|| {
+                matches
+                    .get_one::<String>("filter-only-euser")
+                    .map(|euser| Filter::EUser(euser.clone()))
+            });
+
+        let filter = match filter {
+            Some(Filter::User(data)) => Some(Filter::User(try_into_uid(data)?)),
+            // TODO: Make sure this working
+            Some(Filter::EUser(data)) => Some(Filter::User(try_into_uid(data)?)),
```

Comment:
I guess it should be `EUser` in both cases:
```suggestion
            Some(Filter::EUser(data)) => Some(Filter::EUser(try_into_uid(data)?)),
```